src/middlewares/validators.js

```diff
@@ -0,0 +1,67 @@
+const { body, validationResult } = require('express-validator');
+
+const userValidationRules = () => [
+  body('email')
+    .notEmpty()
+    .withMessage('Email should not be empty')
+    .bail()
+    .isEmail()
+    .withMessage('Not valid email')
+    .bail()
+    .normalizeEmail(),
+  body('username')
+    .notEmpty()
+    .withMessage('Username should not be empty')
+    .bail()
+    .isLength({ min: 3, max: 30 })
+    .withMessage('Username must be more than three and less than 30 characters')
+    .matches(/^[a-zA-Z0-9._]*$/)
+    .withMessage(
+      'Username can contain only alphanumeric characters, period and underscore'
+    )
+    .trim(),
```

Comment:
For the username, I think we can also add a validation to check if it has spaces between letters. Here, `trim()` only checks for the spaces from left and right side of the username. You can write a custom validation for that.

Reply:
<img width="639" alt="image" src="https://user-images.githubusercontent.com/91065358/154988995-c61f4d92-5b20-490e-914a-61adb5b5ad83.png">

here by `.matches(/^[a-zA-Z0-9._]*$/)` I am only allowing lowercase and uppercase alphabet, numbers, and from characters just period and underscore, so considering whitespace is also a character, it will give an error if user includes whitespaces